sources/packages/backend/apps/api/src/route-controllers/student/student.controller.service.ts

```diff
@@ -64,25 +71,57 @@ export class StudentControllerService {
     groupName: string,
     auditUserId: number,
   ): Promise<FileCreateAPIOutDTO> {
-    const createdFile = await this.fileService.createFile(
-      {
-        fileName: file.originalname,
-        uniqueFileName: uniqueFileName,
-        groupName: groupName,
-        mimeType: file.mimetype,
-        fileContent: file.buffer,
-      },
-      studentId,
-      auditUserId,
-    );
-
-    return {
-      fileName: createdFile.fileName,
-      uniqueFileName: createdFile.uniqueFileName,
-      url: `student/files/${createdFile.uniqueFileName}`,
-      size: createdFile.fileContent.length,
-      mimetype: createdFile.mimeType,
-    };
+    let uploadStatus: number;
```

Comment:
The method is organized in a way that makes it hard to read due to the amount of nested if/else.
Please follow the comments and try to use early returns.